### PR TITLE
catch QhullError and return empty array

### LINKED ISSUE #5363: Convex Hull - dependent features can trip up regionprops calculation
## Description

For some degenerate objects (e.g. intrinisically one less dimension, too few voxels), convex hull calculation throws exceptions.
This has been reported by Yi (@sunyi000 on Github) here 
 https://forum.image.sc/t/scikit-image-regionprops-and-regionprops-table-questions/52021/2
for intrinsically 1d objects. In this scase scipy throws the following exception. @jni has also been part of that image.sc thread.

```
scipy.spatial.qhull.QhullError: QH6013 qhull input error: input is less than 3-dimensional since all points have the same x coordinate    0
```
Other objects that can cause such exceptions are those with too few points, see for example:
https://github.com/scikit-image/scikit-image/issues/4865#issuecomment-711069382

## Way to reproduce
Still working on a minimal reproducible example

## Suggested way to fix

IMO,
rather than having `measure.regionprops` fail for such object, `QHullErrors` should be caught and the affected label should have a `NaN` in the relevant features. A warning should also be issued if this occurs.

### PR DESCRIPTION
**Pull request by boazmohar**
## Description
Fixes #5363, for now I am returning a zero image which is also the case if the input image is all zeros.
If nan is preferred I can switch that up.
A test was added that evokes a QhullError.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/morphology/convex_hull.py
Comment (grlee77) [inline skimage/morphology/convex_hull.py:124]:
```suggestion
                warn(f"Failed to get convex hull image"
                     f"Returning empty image, see error message below\n"
                     f"{err}")
```
fix typos and use f-strings

File: skimage/morphology/convex_hull.py
Comment (grlee77) [inline skimage/morphology/convex_hull.py:143]:
```suggestion
        warn(f"Failed to get convex hull image"
             f"Returning empty image, see error message below\n"
             f"{err}")
```

**Review by grlee77 — CHANGES_REQUESTED** [review]
Thanks, @boazmohar. a couple of minor suggestions.

I haven't yet thought about this enough in the context of other properties to determine whether we want to return a NaN image.

File: skimage/morphology/convex_hull.py
Comment (mkcor) [inline skimage/morphology/convex_hull.py:122]:
```suggestion
                warn(f"Failed to get convex hull image. "
```
otherwise it prints `Failed to get convex hull imageReturning empty image, see error message below`

File: skimage/morphology/convex_hull.py
Comment (mkcor) [inline skimage/morphology/convex_hull.py:141]:
```suggestion
        warn(f"Failed to get convex hull image. "
```

File: skimage/morphology/convex_hull.py
Comment (mkcor) [inline skimage/morphology/convex_hull.py:123]:
```suggestion
                     f"Returning empty image, see error message below:\n"
```

File: skimage/morphology/convex_hull.py
Comment (mkcor) [inline skimage/morphology/convex_hull.py:142]:
```suggestion
             f"Returning empty image, see error message below:\n"
```

**Review by mkcor — COMMENTED** [review]
Thanks! Note that you can apply suggestions as a batch: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

**Review by boazmohar — COMMENTED** [review]
Thanks @mkcor, both for the quick review and teaching me about this new batch feature!

**Review by jni — APPROVED** [review]
Thanks @boazmohar!

**Comment by jni** [comment]
@grlee77 my preference is 0s over nans, OR, just return the original image. The rationale for the latter is that one would assume an invariant: the convex hull image should have at least as many "on" pixels as the original. But violating that invariant might be a good way to "soft warn" users that the image is pathological...

File: skimage/morphology/convex_hull.py
Comment (grlee77) [inline skimage/morphology/convex_hull.py:5]: Before we merge this, let me double check if this `QhullError` is public API. SciPy has recently been renaming a lot of files to have underscores that were not intentionally public in the past...

Comment (grlee77) [reply skimage/morphology/convex_hull.py:5]: `scipy.spatial.qhull` is not public API as listed here:
https://docs.scipy.org/doc/scipy-1.7.0/reference/#api-definition
In fact it has now been renamed to `scipy.spatial._qhull` and unfortunately the `__all__` attribute on _qhull.pyx [does not include `QhullError`](https://github.com/scipy/scipy/blob/master/scipy/spatial/_qhull.pyx#L31).  Let me check with SciPy about this...

Comment (grlee77) [reply skimage/morphology/convex_hull.py:5]: I opened https://github.com/scipy/scipy/issues/14999 upstream to ask about this

Comment (boazmohar) [reply skimage/morphology/convex_hull.py:5]: > 
> 
> I opened [scipy/scipy#14999](https://github.com/scipy/scipy/issues/14999) upstream to ask about this


[scipy/scipy#15003](https://github.com/scipy/scipy/pull/15003) is now merged into main that would make `QhullError` public API, great catch @grlee77 

Let me know if I need to do anything else.

**Comment by boazmohar** [comment]
Now that [scipy/scipy#15003](https://github.com/scipy/scipy/pull/15003) is merged I see 2 options:
1. Change to `from scipy.spatial import QhullError` and the test would fail until a new scipy is released.
2. Leave as `from scipy.spatial.qhull import QhullError` and merge now (would give a `DeprecationWarning`) and do another PR when scipy is released.

I don't have a preference, @jnl, @grlee77, @mkcor  any thoughts?

**Comment by grlee77** [comment]
> I don't have a preference, @jnl, @grlee77, @mkcor any thoughts?

I would recommend the following which should first try the upcoming SciPy public API and then fall back to the unofficial one otherwise.

```Python
try:
    from scipy.spatial import QhullError
except ImportError:
    from scipy.spatial.qhull import QhullError
```

**Comment by grlee77** [comment]
It may also be helpfull to add a line in the `TODO.txt` file in the root of the repository reminding us to go back and remove the except clause a couple of years in the future once SciPy 1.8 becomes our minimum dependency.

File: skimage/morphology/convex_hull.py
Comment (grlee77) [reply skimage/morphology/convex_hull.py:5]: Thank you for quickly making the upstream PR @boazmohar!

File: TODO.txt
Comment (mkcor) [inline TODO.txt:70]:
```suggestion
* When ``scipy`` is set to >= 1.8.0, remove conditional import of QhullError in
  ``skimage/morphology/convex_hull.py``.
```

**Review by grlee77 — APPROVED** [review]
The updates here look good to me. Thanks @boazmohar. The CI failures seem to be something unrelated (sphinx gallery)

**Comment by jni** [comment]
> The CI failures seem to be something unrelated (sphinx gallery)

*sigh* new Sphinx release today. Let's merge this and then fix that in another PR. Thank you @boazmohar!

**Pull request merged**
